sys/arm64/cheri/cheri_machdep.c

```diff
@@ -89,3 +89,26 @@ hybridabi_thread_setregs(struct thread *td, unsigned long entry_addr)
 	    td, CHERI_CAP_USER_CODE_PERMS, CHERI_CAP_USER_CODE_BASE,
 	    CHERI_CAP_USER_CODE_LENGTH, entry_addr));
 }
+
+/*
+ * As with system calls, handling signal delivery connotes special authority
+ * in the runtime environment.  In the signal delivery code, we need to
+ * determine whether to trust the executing thread to have valid stack state,
+ * and use this function to query whether the execution environment is
+ * suitable for direct handler execution, or if (in effect) a security-domain
+ * transition is required first.
+ */
+int
+cheri_signal_sandboxed(struct thread *td)
+{
+#if 0
```

Comment:
Is this needed?

Reply:
That is a good question.. for @rwatson.  We've kept this code (enabled) for RISC-V using SW0 as the syscall permission.   I think we do have SYSCALL as the SW0 permission on Morello, so we could probably enable this (I will try it in a bit.)  I think the question of if we need a SYSCALL software permission is cross-platform though.

Reply:
Debatable. You'd either want to include it or delete the code (`#if 0` is pointless when it's just copied from RISC-V). I don't think there's any harm having it though, it just isn't really used because nothing clears the bit except the MIPS-specific libcheri.

Reply:
(I think either you should make it a MIPS-specific thing and GC the RISC-V equivalent, or keep it an MI concept)

Reply:
Actually, the checks for it are in sys/mips because it's about signal delivery. We should just delete it from arm64 and riscv and move the prototype out of sys/cheri into sys/mips.

Reply:
I've enabled it (and trimmed a no-longer-necessary cast).  cheribsdtest still passes, but I suspect the only real test for this is in the libcheri tests that we don't run on non-MIPS.

Reply:
I would be fine making it MIPS-only.  I think doing it as a followup change might simplify merging back to morello-dev as doing a merge will then remove it from morello-dev as well (assuming we do a merge of this PR first).